### DISCUSSION TITLE
Recommended use for application specific pages and components

### DISCUSSION THREAD

**Discussion by norcino** [storybookjs/storybook#17406] [Help]
Hi there, I been using storybook successfully for a few months now, it is principally used to render, document and test a component library I started setting up last November.
Personally, beside the initial troubles and effort needed to set everything up as I needed, I actually have no problem with it, I am still getting familiar with the "mocking" of the dependencies but beside that I feel comfortable using storybook.

I would like to know if you use storybook also for applications where components are not meant to be reused, where you have bespoke pages where you implement your website.
If you do use it can you advice if there is any pitfall to avoid, or any other recommendation to follow?
From a RoI point of view, do you think it is worth the creation of stories for specific web app?

In my view, the way I setup storybook, the usage in all web apps would be a good practice, but pretty much all people I came across with experience or an idea about storybook, they would have used only for component libraries.

The benefits I have identified are:
- writing stories as you create components helps you design better components, more loosely couple (me and my team come more from a OOP paradigm, and functional programming is not our main skill)
- using snapshot testing will facilitate the detection of UI issues that would be discoverable only if a QA look at the screen, and still minor changes might escape review, causing the UI to drift away from the original UX vision
- we can use storybook to test internationalization
- RTL can be easily applied to any component
- would represent a valuable internal documentation for onboarding, maintenance and KT
- it can be used for accessibility testing
- would help development as platform where components can be rendered and customized in real time and api dependencies can be lifted
- I also believe that the creation of stories can help also with the creation of tests

The cost of the adoption would be related to:
- writing the documentation/control properties
- design of the individual stories
- wire up the mocking of the dependencies of the component (this can help with the creation of the tests though)

I look forward to read about your experience and opinion.

**Comment by Nushaine** [comment]
@norcino did your team end up going through with it? i'm considering adopting storybook in a similar manner for our frontend app (outside of our internal component library)
